src/wovnio/wovnphp/Diagnostics.php

```diff
@@ -120,14 +134,22 @@ private function initResults()
         );
     }
 
-    private static function getWovnIni()
+    private function getWovnIni()
     {
-        $settingFileName = isset($env['WOVN_CONFIG']) ? $env['WOVN_CONFIG'] : DIRNAME(__FILE__) . '/../../../../wovn.ini';
-
-        if (file_exists($settingFileName)) {
-            $userSettings = parse_ini_file($settingFileName, true);
+        if (isset($this->env['WOVN_CONFIG'])) {
+            $settingFileName = $this->env['WOVN_CONFIG'] ? $this->env['WOVN_CONFIG'] : DIRNAME(__FILE__) . '/../../../../wovn.json';
+            if (file_exists($settingFileName)) {
+                $userSettings = file_get_contents($settingFileName);
+            } else {
+                $userSettings = array('Error' => 'Not Found');
+            }
         } else {
-            $userSettings = array('Error' => 'Not Found');
+            $settingFileName = DIRNAME(__FILE__) . '/../../../../wovn.ini';
+            if (file_exists($settingFileName)) {
+                $userSettings = file_get_contents($settingFileName);
+            } else {
+                $userSettings = array('Error' => 'Not Found');
+            }
```

Comment:
I think code like the following is more simple.

```
if (isset($this->env['WOVN_CONFIG'])) {
  $settingFileName = $this->env['WOVN_CONFIG'] ? $this->env['WOVN_CONFIG'] : DIRNAME(__FILE__) . '/../../../../wovn.json';
} else {
  $settingFileName = DIRNAME(__FILE__) . '/../../../../wovn.ini';
}

if (file_exists($settingFileName)) {
  $userSettings = file_get_contents($settingFileName);
} else {
  $userSettings = array('Error' => 'Not Found');
}
```

Reply:
Good point! What I wrote was definitely very spaghetti like.